homeassistant/components/tplink/__init__.py

```diff
@@ -86,11 +129,40 @@ async def _async_discovery(*_: Any) -> None:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     host = entry.data[CONF_HOST]
+
+    credentials = await get_stored_credentials(hass)
+    connection_type = None
+
+    if connection_type_dict := entry.data.get(CONF_CONNECTION_TYPE):
+        try:
+            connection_type = ConnectionType.from_dict(connection_type_dict)
+        except SmartDeviceException:
+            _LOGGER.warning(
+                "Invalid connection parameters for %s: %s", host, connection_type_dict
+            )
+
+    httpx_asyncclient = create_async_httpx_client(hass, verify_ssl=False)
```

Comment:
We should avoid creating an httpx client if the device doesn't need it. The bulk of users are not going to have the new devices yet

Reply:
Fixed, will now only create for devices that use http